### DISCUSSION TITLE
Help building your own USB/IP enabled WSL 2 kernel

### DISCUSSION THREAD

**Discussion by jjb7733** [dorssel/usbipd-win#404] [Q&A]
I am following the support topic for building your own USB/IP enabled WSL 2 kernel [https://github.com/dorssel/usbipd-win/wiki/WSL-support#building-your-own-usbip-enabled-wsl-2-kernel](url), I got all the way to the end and stuck at this part:

From the root of the repo, copy the image.
`cp arch/x86/boot/bzImage /mnt/c/Users/<user>/usbip-bzImage`
cp: cannot stat 'arch/x86/boot/bzImage': No such file or directory

I have searched the directory the bzImage file does not exist in the directory arch/x86/boot/bzImage, not sure where I went wrong, where does this file come from or in which step is it created so that I can figure out where I went wrong?

**Comment by dorssel** [comment]
It is (or rather: should have been) created by
```
sudo make -j 8 && sudo make modules_install -j 8 && sudo make install -j 8
```
Did that line give any errors? Somebody else already reported that the instructions later say `cd tools/usb/usbip` but then even later it says

> From the root of the repo, copy the image.

This means you have to go up three levels again...

**Reply by Shadouken** [reply]
Running into the same issue. No errors thrown and going through the process it seems that bzImage was never installed. Running linux-msft-wsl-5.10.102.1

**Reply by dorssel** [reply]
What distro are you using for building? Procedure is for Ubuntu 20.04.

**Reply by Shadouken** [reply]
Using 20.04

**Reply by dorssel** [reply]
Did you enable any additional drivers (I guess so...)? Some drivers need additional tooling (packages) installed to compile. The tools in the Wiki instruction are only enough to compile the default WSL2 configuration. On the other hand, I would have expected at least one error if any such additional tooling was missing. All I can say is that many people (including myself) have compiled the kernel following exactly those Wiki instructions...
